src/reporter.ts

```diff
@@ -442,6 +447,30 @@ export default class Reporter {
           data: fs.createReadStream(s.path),
         });
       });
+
+      if (
+        this.opts.artifactUploadDir &&
+        fs.existsSync(this.opts.artifactUploadDir)
+      ) {
+        const artifactPath = path.join(this.opts.artifactUploadDir, specName);
```

Comment:
I think you have to do another `fs.exists` check here. The user may not have assets for every single spec, so the directories may legitimately not exist.

Reply:
derp, good call.